### PR TITLE
Use ifo_list and string as section tags

### PR DESCRIPTION
**Pull request by spxiwh**
This changes the behaviour of jobs when using ifo_list=['H1','L1','V1'] so that options in [executable-H1], [executable-L1], [executable-V1] and executable-H1L1V1 are picked up. (The last may not be meaningful, and may be removed in the future).

This is needed for the O1 GRB runs and should be considered for the 1.2.5 release.

### CODE REVIEW

**Comment by spxiwh** [comment]
This should address #540. @a-r-williamson can you test that this works? @ahnitz I can't see that this will cause a problem anywhere else, as I don't think any of the all-sky codes use ifo_list=['H1','L1','V1'] instead of ifo_list=['H1L1V1'] and options from [executable-H1L1V1] are still read even if they are.

File: pycbc/workflow/core.py [hunk 1]
```diff
@@ -213,8 +213,8 @@ def __init__(self, cp, name,
         # Determine the sections from the ini file that will configure
         # this executable
         sections = [name]
-        if self.ifo_string:
-            sec_tags = tags + [self.ifo_string]
+        if self.ifo_list is not None
+            sec_tags = tags + [self.ifo_list] + [self.ifo_string]
```
Comment (a-r-williamson) [inline pycbc/workflow/core.py]: The brackets around self.ifo_list should be removed, it is already a list and will raise an error. There is then an issue for single IFO running that self.ifo_list and [self.ifo_string] are identical, so we see the error:

>File "/home/c1239174/pycbc-dev/lib/python2.6/site-packages/PyCBC-88f5e1-py2.6.egg/pycbc/workflow/configuration.py", line 516, in check_duplicate_options
>    %(section1,section2,' '.join(duplicates)))
>ValueError: The following options appear in both section inspiral-h1 and inspiral-h1: h1-channel-name

I'd propose something more like:
```python
if self.ifo_list is not None:
    if len(self.ifo_list) > 1:
        sec_tags = tags + self.ifo_list + [self.ifo_string]
    else:
        sec_tags = tags + self.ifo_list
else:
    sec_tags = tags
```

**Comment by spxiwh** [comment]
Thanks Andrew. I've updated the patch, let me know if this is good now.

**Comment by a-r-williamson** [comment]
closes #540

**Pull request merged**